### ISSUE TITLE
Input problems: lags after pressing the spacebar, unicode input

### ISSUE THREAD

**Issue by nkaretnikov** [derekelkins/agda-vim#12]
Agda version 2.4.2.4
VIM - Vi IMproved 7.4

I can't figure out how to quickly type text while use agda-vim. In addition to that, I'm having trouble with unicode input.

```
let mapleader      = ","
let maplocalleader = ","
```

For example, if I try to type
```
if_then_else_ : {A : Set} → Bool → A → A → A
```
normally (like I'd do in the usual Insert mode), I'll get something like this:
```
if_then_else_∶ {A∶Σet}→> Bool→> A→> A→> A
```
(the arrows are inserted as `->` here).

It seems I need to wait for something to timeout to type normally, so if I try the same slowly while waiting a few seconds after each space, I get this:
```
if_then_else_ : {A : Set} -> Bool -> A -> A -> A
```

This looks better, but still unacceptable for normal usage. And the arrows are in plain text. Hitting space twice works sometimes, but it also may actually insert two spaces, e.g., between `:` and `{A`:
```
if_then_else_ ∶  {A
```

So my current way of doing it is to use double space while keeping in mind that the space character is inserted automatically before `{` and after `\to`. I still need to wait a bit while typing `Set` (or I'd get `Σet`), though. Can I do better?

**Comment by nkaretnikov** [comment]
Also, unicode input works in the `:`-command buffer, is it really needed? Can it be disabled?

**Comment by nkaretnikov** [comment]
Okay, I've just learned the hard way that typing space twice may cause problems. This file looks right, but I hit double space before entering some colons, so the colons here are not the same, which leads to a parse error:

```
data ℕ ∶ Set where
  zero : ℕ
  succ ∶ ℕ → ℕ

... Parse error
Set<ERROR>
 where
  zero : ℕ
  succ ∶ ℕ →...
```

**Comment by nkaretnikov** [comment]
I think I got it. I was using the space key as the leader key before configuring agda-vim. So I changed that to `,` to be consistent with the local leader, but I did that without exiting vim (by using `:so %`). It seems that something got messed up as the result.

The bottom line is that I now have both leaders set to `,`, and everything works as expected. I can insert an arrow with `,to` for instance.
